python/lancedb/table.py

```diff
@@ -746,11 +843,12 @@ def create_index(
             accelerator=accelerator,
             index_cache_size=index_cache_size,
         )
-        self._reset_dataset()
```

Comment:
Was this meant to go away?

Reply:
Yes, I've removed reset_dataset intentionally. It's unnecessary.